### PR TITLE
Func read file record

### PR DESCRIPTION
**Pull request by NVTaranets**
# Function "read file record"  
I "read file record" function implemented.
Due to the specifics of the .execute parameters
**starting_address, quantity_of_x, number_file** must be **tuple ()** of one long (by the number of requested sub_seq) 
the **result** will be **((sub _ seq_0 _ data), (sub_seq_1_data),... (sub_seq_N_data))**.
The implementation has been validated on a single manufacturer's real devices.

### CODE REVIEW

File: modbus_tk/modbus.py [hunk 1]
```diff
@@ -338,6 +365,13 @@ def execute(
                             digits.append(byte_val % 2)
                             byte_val = byte_val >> 1
                     result = tuple(digits)
+                if function_code == defines.READ_FILE_RECORD:
+                    sub_seq=list()
```
Comment (ljean) [inline modbus_tk/modbus.py:369]: The code should follow the coding style of the project and have one space between operators

For examplle : `ptr < len(results)`, `sub_seq += ((ptr + 2, ptr + 2 + results[ptr] // 2`, `ptr+=result[ptr] // 2 + 2`, `result=tuple(map(lambda x: result[x[0]:x[1]], sub_seq))`

varibles should be at lealt 3 letters (x in lambda is not correct). The name should be "explicative"

File: modbus_tk/modbus.py [hunk 2]
```diff
@@ -177,6 +182,28 @@ def execute(
...
+                    and isinstance(quantity_of_x, tuple)
+                    and len(number_file)==len(starting_address)==len(quantity_of_x)>0
+            ):
+                count_seq=len(number_file)
+            else:
+                raise ModbusInvalidRequestError(
+                    'For function READ_FILE_RECORD param'
+                    'starting_address, quantity_of_x, number_file must be tuple ()'\
+                     'of one length > 0 (by the number of requested sub_seq)'
+                     )
```
Comment (ljean) [inline modbus_tk/modbus.py:198]: indentation of closing parenthesis should be at same level than raise

File: modbus_tk/modbus.py [hunk 3]
```diff
@@ -177,6 +182,28 @@ def execute(
                 # slave + func + bytcodeLen + bytecode x 2 + crc1 + crc2
                 expected_length = 2 * quantity_of_x + 5
 
+        elif function_code == defines.READ_FILE_RECORD:
+            is_read_function = True 
+            if (isinstance(number_file, tuple)
```
Comment (ljean) [inline modbus_tk/modbus.py:187]: New line after opening parenthesis. first condition should be at the same level than others.

File: modbus_tk/modbus.py (hunk 1 above)
Comment (NVTaranets) [reply modbus_tk/modbus.py]: Accepted, resolved in the next commits

File: modbus_tk/modbus.py (hunk 2 above)
Comment (NVTaranets) [reply modbus_tk/modbus.py]: Accepted, resolved in the next commits

File: modbus_tk/modbus.py (hunk 3 above)
Comment (NVTaranets) [reply modbus_tk/modbus.py]: Accepted, resolved in the next commits

**Review by ljean — APPROVED** [review]

**Pull request merged**